### PR TITLE
use fused RoPE kernel in DeepseekScalingRotaryEmbedding

### PR DESCRIPTION
**Pull request by yangulei**
Remove `DeepseekScalingRotaryEmbedding.forward()` to re-use `RotaryEmbedding.forward_hpu()` instead. The `RotaryEmbedding.forward_hpu()` calls [apply_rotary_pos_emb](https://docs.habana.ai/en/v1.21.1/PyTorch/Reference/Python_Packages.html#hpex-kernels-rotaryposembeddinghelper) for better performance.

Before:
image

After:
image

### CODE REVIEW

**Comment by yangulei** [comment]
@xuechendi @kwisniewski98 
Please help to review, thanks!

File: vllm/model_executor/layers/rotary_embedding.py [hunk 1]
```diff
@@ -869,7 +869,7 @@ def _compute_cos_sin_cache(self) -> torch.Tensor:
         cache = torch.cat((cos, sin), dim=-1)
         return cache
 
-    def forward(
+    def forward_native(
```
Comment (kwisniewski98) [inline vllm/model_executor/layers/rotary_embedding.py:872]: Is it necessary to rename this function?

File: vllm/model_executor/layers/rotary_embedding.py [hunk 2]
```diff
@@ -909,6 +909,49 @@ def forward(
             key = key_rot
         return query, key
 
+    def forward_hpu(
```
[the remark this answers is not in the document]
Comment (yangulei) [reply vllm/model_executor/layers/rotary_embedding.py:912]: I'm trying to delete `forward()` and reuse `forward_hpu()` in `RotaryEmbedding`, it will be nice if it works, thanks!

[the remark this answers is not in the document]
Comment (yangulei) [reply vllm/model_executor/layers/rotary_embedding.py]: Done.

vllm (pretrained=/models/DeepSeek-R1-G2,tensor_parallel_size=8,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=4096,use_v2_block_manager=True,dtype=bfloat16,device=hpu), gen_kwargs: (None), limit: None, num_fewshot: 5, batch_size: 8
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9553|±  |0.0057|
|     |       |strict-match    |     5|exact_match|↑  |0.9560|±  |0.0056|

File: vllm/model_executor/layers/rotary_embedding.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def forward_hpu(
         if hasattr(self, "scaling_factors") or hasattr(
                 self, "scaling_factor") or self.sin is None:
             self.prepare_cos_sin(positions, offsets)
-        num_tokens = positions.shape[0] * positions.shape[1]
+        num_tokens = positions.numel()
```
Comment (xuechendi) [inline vllm/model_executor/layers/rotary_embedding.py:265]: any specific reason to update this?

File: vllm/model_executor/layers/rotary_embedding.py [hunk 4]
```diff
@@ -869,46 +869,6 @@ def _compute_cos_sin_cache(self) -> torch.Tensor:
         cache = torch.cat((cos, sin), dim=-1)
         return cache
 
-    def forward(
-        self,
-        positions: torch.Tensor,
-        query: torch.Tensor,
-        key: torch.Tensor,
-        offsets: Optional[torch.Tensor] = None,
-    ) -> Tuple[torch.Tensor, torch.Tensor]:
```
[the remark this answers is not in the document]
Comment (yangulei) [reply vllm/model_executor/layers/rotary_embedding.py:878]: The `forward()` have the highest priority, then `forward_<platform_name>`. So, we could rename the `forward()` to `forward_native()` or just delete it. I prefer to remove `DeepseekScalingRotaryEmbedding.forward()` as its logic is identical to `RotaryEmbedding.forward_native()`.

File: vllm/model_executor/layers/rotary_embedding.py (hunk 3 above)
Comment (yangulei) [reply vllm/model_executor/layers/rotary_embedding.py:265]: The `positions` of Deepseek-R1 is a flattened 1D tensor.
Using `numel()` is much robust and aligned with `forward_native()`:
https://github.com/HabanaAI/vllm-fork/blob/deb019dcf767d012434f1f301547c95fdbf7be9a/vllm/model_executor/layers/rotary_embedding.py#L175-L176
